File: account-service/src/main/java/accountservice/model/Account.java
package accountservice.model;

import accountservice.model.enums.Role;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String username;
    private String email;
    private String password;
    private Role role;
    @CreationTimestamp
    private Timestamp createdAt;
    @UpdateTimestamp
    private Timestamp updatedAt;
}

File: account-service/src/main/java/accountservice/model/Account.kt
package accountservice.model

import accountservice.model.enums.Role
import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.sql.Timestamp
import java.util.UUID


@Entity
data class Account @JvmOverloads constructor(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,
    @Column(unique = true)
    var username: String,
    var email: String,
    val password: String,
    @Enumerated(EnumType.ORDINAL)
    val role: Role,
    @CreationTimestamp
    val createdAt: Timestamp? = null,
    @UpdateTimestamp
    val updatedAt: Timestamp? = null
)
